### PR TITLE
Roberta data reader

### PR DESCRIPTION
**Pull request by jacob-morrison**
Adding a dataset reader and training config for RoBERTa on SuperGLUE RTE

### CODE REVIEW

File: allennlp_models/pair_classification/dataset_readers/transformer_superglue_rte.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        }
+
+    @overrides
+    def _read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path, extract_archive=True)
+
+        logger.info("Reading file at %s", file_path)
+        yielded_relation_count = 0
+        for relation in self.shard_iterable(jsonlines.open(file_path)):
```
Comment (dirkgr) [inline allennlp_models/pair_classification/dataset_readers/transformer_superglue_rte.py:65]: Use https://github.com/allenai/allennlp/blob/main/allennlp/common/file_utils.py#L952, so we don't need a new dependency.

File: training_config/pair_classification/superglue_rte_roberta.jsonnet [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      "type": "huggingface_adamw",
+      "weight_decay": 0.01,
+      "parameter_groups": [[["bias", "LayerNorm\\.weight", "layer_norm\\.weight"], {"weight_decay": 0}]],
+      "lr": 1e-6,
+      "eps": 1e-8,
+      "correct_bias": true
+    },
+    "learning_rate_scheduler": {
+      "type": "linear_with_warmup",
+      "warmup_steps": 100
```
Comment (dirkgr) [inline training_config/pair_classification/superglue_rte_roberta.jsonnet:56]: That's not a lot of warmup, but if you're getting the numbers, who am I to argue?

**Review by dirkgr — CHANGES_REQUESTED** [review]
Only small changes requested

File: allennlp_models/pair_classification/dataset_readers/transformer_superglue_rte.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            premise = relation["premise"]
+            hypothesis = relation["hypothesis"]
+            if "label" in relation:
+                label = relation["label"]
+            else:
+                label = None
+            index = relation["idx"]
+
+            # todo: see if we even need this to be in a separate method
+            instance = self.make_instance(index, label, premise, hypothesis)
```
[the remark this answers is not in the document]
Comment (jacob-morrison) [reply allennlp_models/pair_classification/dataset_readers/transformer_superglue_rte.py:75]: Done

File: training_config/pair_classification/superglue_rte_roberta.jsonnet (hunk 2 above)
Comment (jacob-morrison) [reply training_config/pair_classification/superglue_rte_roberta.jsonnet:56]: ¯\_(ツ)_/¯

File: allennlp_models/modelcards/pair-classification-roberta-rte.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{
+    "id": "pair-classification-roberta-rte",
+    "registered_model_name": "roberta-rte",
+    "registered_predictor_name": null,
+    "display_name": "RoBERTa RTE",
+    "task_id": "pair_classification",
+    "model_details": {
+        "description": "The model implements a pair classification model patterned after the proposed model in [BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding (Devlin et al, 2018)](https://api.semanticscholar.org/CorpusID:52967399), fine-tuned on the MultiNLI corpus. It predicts labels with a linear layer on top of word piece embeddings.",
+        "short_description": "A pair classification model patterned after the proposed model in Devlin et al, fine-tuned on the MultiNLI corpus",
```
[the remark this answers is not in the document]
Comment (jacob-morrison) [reply allennlp_models/modelcards/pair-classification-roberta-rte.json:9]: Ah yeah fair point. I put MNLI because I used roberta-large-mnli like FB did, but I then technically finetuned that on RTE.

**Review by dirkgr — APPROVED** [review]
